iceoryx_posh/include/iceoryx_posh/internal/popo/sample.inl

```diff
@@ -22,96 +23,27 @@ namespace iox
 {
 namespace popo
 {
-namespace internal
-{
-template <typename T, typename H>
-inline SamplePrivateData<T, H>::SamplePrivateData(cxx::unique_ptr<T>&& sampleUniquePtr,
-                                                  PublisherInterface<T, H>& publisher) noexcept
-    : sampleUniquePtr(std::move(sampleUniquePtr))
-    , publisherRef(publisher)
-{
-}
-
-template <typename T, typename H>
-inline SamplePrivateData<const T, H>::SamplePrivateData(cxx::unique_ptr<const T>&& sampleUniquePtr) noexcept
-    : sampleUniquePtr(std::move(sampleUniquePtr))
-{
-}
-} // namespace internal
-
 template <typename T, typename H>
 template <typename S, typename>
 inline Sample<T, H>::Sample(cxx::unique_ptr<T>&& sampleUniquePtr, PublisherInterface<T, H>& publisher) noexcept
-    : m_members({std::move(sampleUniquePtr), publisher})
+    : SmartChunk<PublisherInterface<T, H>, T, H>(std::move(sampleUniquePtr), publisher)
```

Comment:
I think it should be possible to use `Base_t` instead of `SmartChunk<PublisherInterface<T, H>, T, H>`